aries_cloudagent/protocols/present_proof/dif/pres_exch_handler.py

```diff
@@ -1187,12 +1206,18 @@ async def create_vp(
     def check_sign_pres(self, creds: Sequence[VCRecord]) -> bool:
         """Check if applicable creds have CredentialSubject.id set."""
         for cred in creds:
-            if len(cred.subject_ids) > 0 and not next(
-                iter(cred.subject_ids)
-            ).startswith("urn:"):
+            if len(cred.subject_ids) > 0 and not self.check_if_cred_id_derived(
+                next(iter(cred.subject_ids))
+            ):
                 return True
         return False
 
+    def check_if_cred_id_derived(self, id: str) -> bool:
+        """Check if credential or credentialSubjet id is derived."""
+        if id.startswith("urn:") and "c14" in id:
```

Comment:
I think you can be more explicit:

```suggestion
        if id.startswith("urn:bnid:_:c14n")
```